### PR TITLE
[Process] Regression: Error output-handling is affecting Drupal core development

### PR DESCRIPTION
**Pull request by TravisCarden**
| Q | A
| -- | ---
| Branch? | 7.1
| Bug fix? | yes
| New feature? | no
| Deprecations? | no
| Issues | n/a
| License | MIT

**tl;dr**: Since https://github.com/symfony/symfony/pull/52409, `\Symfony\Component\Process\Process` no longer captures error output when attempting to run a non-existent command on Linux--behavior Drupal core was relying on. The regression was first released in [v7.1.0-BETA1](https://github.com/symfony/symfony/releases/tag/v7.1.0-BETA1) and persists into `7.2`.

### Details

https://github.com/symfony/symfony/pull/52409 changed how the commandline is passed to `proc_open()`, apparently triggering a bug in PHP (https://github.com/php/php-src/issues/12589) whereby invoking a non-existent command on Linux no longer provides any error output. That's important behavior for a major error condition. Hence my surprise when I discovered that it was a known issue before the PR was merged--that it was explicitly called out and that tests were simply updated to avoid it.

As a consequence of the change, tests started failing on Drupal core component [Composer Stager](https://github.com/php-tuf/composer-stager). This has caused a ripple effect and taken me away from other Drupal core work. I would _really_ like to see it fixed quickly.

### Steps to Reproduce

This PR **demonstrates** the problem. It does not yet fix it. (Since the regression was part of a performance enhancement, not a functional change, perhaps it wouldn't be out of the question to just revert it, if necessary.) This begins by temporarily reverting the regression and adding a test that we can see passing on CI<sup>[1]</sup>. Then it puts the regression back in place so we can see it failing.

This...
```php
$p = new Process(['invalid_command']);
$p->run();
var_dump($p->getErrorOutput());
```

🟢 Used to print this on Linux:
```
string(45) "sh: line 0: exec: invalid_command: not found
"
```

🔴 Now it prints this:
```
string(0) ""
```

### To-do
- [ ] Fix the regression
- [ ] Ensure adequate test coverage
- [ ] Update `src/**/CHANGELOG.md` files

---

<sup>[1]</sup> Well, not very obviously, I guess, since a bunch of tests were already failing.

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Reverting #52409 would be a step backward. What I don't understand from the description is how this is affecting you?
Were you relying on the details of the error output? If yes, then I'd argue this is too fragile and I would seek for an alternative way to achieve what you do with this sniffing. If not, then please expand.

**Comment by nicolas-grekas** [comment]
Using `new Process(['sh', '-c', 'invalid_command']);` might be a way forward for you.
Using the PhpExecutableFinder might be another one, depending on what you need.

**Comment by nicolas-grekas** [comment]
Let me close here since we won't merge this patch.

**Comment by TravisCarden** [comment]
Thanks, @nicolas-grekas. Just for a little parting clarity as we close this, I wasn't depending on _what_ the output was. I was depending on the fact that there _was_ output. When I initially filed this, I thought `Process:: getErrorOutput()` had just plain stopped working. Only as I continued debugging did I discover the more limited scope of the problem, i.e., it only occurs when the invoked command doesn't exist. At least it's clearly documented now, in case someone else runs into the same problem. Thanks.
